contracts/signal_oracle.py

```
@@ -0,0 +1,228 @@
+            approval_threshold=threshold,
+        )
+
+    @gl.public.write
+    def verify_web_signal(self, headline: str, source_url: str) -> dict:
+        """
+        Verify a signal by checking its source URL.
+        Uses GenLayer's web access to validate claims against source material.
+        UNIQUE TO GENLAYER: No other blockchain can do this.
+        """
+        def verify_against_source() -> str:
+            web_content = gl.nondet.web.render(source_url, mode="text")
```

Ruff (0.15.7)

[error] `gl` may be undefined, or defined from star imports

(F405)